File: DSA/Sorting/C/insertion.c
#include<stdio.h>
void selection(int arr[],int num){
    for (int i = 1; i < num; i++)
    {
        int temp=arr[i];
        int j=i-1;
        while (j >= 0 && arr[j] > temp)
        {
            arr[j+1]=arr[j];
            j-=1;
        }
        arr[j+1]=temp;
    }
    for (int i = 0; i < num; i++)
    {
        printf("%d ",arr[i]);
    }
    
    
}
int main(){
    int num;
    printf("Enter the Length: ");
    scanf("%d",&num);
    int arr[num];
    for (int i = 0; i < num; i++)
    {
        scanf("%d",&arr[i]);
    }
    
    selection(arr,num);

    return 0;
}

File: DSA/Sorting/Python/insertion.py
def insertion(arr):
    n=len(arr)
    for i in range(1,n):
        temp=arr[i]
        j=i-1
        while(j>=0 and arr[j]>temp):
            arr[j+1]=arr[j]
            j-=1
        arr[j+1]=temp
    return arr            
            
if __name__=="__main__":
    inp=list(map(int,input("Enter the Elements : ").split()))
    ans=insertion(inp)
    print(ans)